### PR TITLE
Cart

### PR DESCRIPTION
**Pull request by Quadry01**

### CODE REVIEW

File: app/LoginPage/page.js [hunk 1]
```diff
@@ -0,0 +1,82 @@
+"use client";
```
Comment (ajaps) [inline app/LoginPage/page.js:1]: [Question]
What's this for?

File: app/LoginPage/page.js [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+  };
+
+  // LOGIN USER WITH EMAIL AND PASSWORD
+  const LoginWithEmail = async (e) => {
+    try {
+      e.preventDefault();
+      await createUserWithEmailAndPassword(auth, email, password);
+      RedirectUser();
+    } catch (error) {
+      console.log(error);
```
Comment (ajaps) [inline app/LoginPage/page.js:23]: [Blocker]
Like I said in the previous PR, research on how to display errors to users.

`console.log` isn't an appropriate way to display errors to users

File: app/LoginPage/page.js [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+  const [password, setPassword] = useState("");
+  // REDIRECT USER AFTER SIGNIN
+  const RedirectUser = (e) => {
+    window.location.replace("ProductPage");
+  };
+
+  // LOGIN USER WITH EMAIL AND PASSWORD
+  const LoginWithEmail = async (e) => {
+    try {
+      e.preventDefault();
```
Comment (ajaps) [inline app/LoginPage/page.js:19]: [Question]
Why do you need this?

File: app/ProductPage/page.js [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+import { collection, getDocs } from "firebase/firestore";
+import { BsCartDash, BsPlusSquare } from "react-icons/bs";
+import { AiOutlineDelete, AiOutlineMinusSquare } from "react-icons/ai";
+import { React, useEffect, useState } from "react";
+
+function ProductsPage() {
+  const [cart, setCart] = useState({});
+  const [hidecart, setHideCart] = useState("hidden");
+  const [productList, setProductList] = useState([]);
+  const values = Object.values(cart);
```
Comment (ajaps) [inline app/ProductPage/page.js:13]: [Question]
why is this necessary?

File: app/ProductPage/page.js [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+import Image from "next/image";
+import { db } from "@/FirebaseConfig/Firebase";
+import { collection, getDocs } from "firebase/firestore";
+import { BsCartDash, BsPlusSquare } from "react-icons/bs";
+import { AiOutlineDelete, AiOutlineMinusSquare } from "react-icons/ai";
+import { React, useEffect, useState } from "react";
+
+function ProductsPage() {
+  const [cart, setCart] = useState({});
+  const [hidecart, setHideCart] = useState("hidden");
```
Comment (ajaps) [inline app/ProductPage/page.js:11]: [Blocker]
Variable names should be camelCase

File: app/ProductPage/page.js [hunk 6]
```diff
@@ -0,0 +1,184 @@
...
+  const addItemToCart = (product) => {
+    setCart({ ...cart, [product.id]: { ...product.data, quantity: 1 } });
+  };
+  // REMOVE ITEM FROM CART
+  const removeItemFromCart = (product) => {
+    setCart(
+      Object.assign(
+        {},
+        Object.values(cart).filter((filterTag) => filterTag !== product)
+      )
```
Comment (ajaps) [inline app/ProductPage/page.js:45]: [Comment]
I will like you to explain to me on call, how this works

File: app/ProductPage/page.js [hunk 7]
```diff
@@ -0,0 +1,184 @@
...
+        const products = res.docs.map((doc) => ({
+          data: doc.data(),
+          id: doc.id,
+        }));
+        setProductList(products);
+      })
+      .catch((err) => {
+        alert(err.message);
+      });
+  };
```
Comment (ajaps) [inline app/ProductPage/page.js:71]: [Suggestion]
I will like you to update this block of code to use async/await

**Review by ajaps — COMMENTED** [review]
Good progress.

I see the improvement in the variable naming 👍 

Can you please provide steps to setup your app locally in your readme.  I'm unable to set it up properly.

e.g. https://github.com/paulsaenzsucre/To-Do-List

File: app/LoginPage/page.js (hunk 1 above)
Comment (Quadry01) [reply app/LoginPage/page.js:1]: This is used for Next.js client side rendering....i was required to add it because i made use of some react hooks on the page

File: app/LoginPage/page.js (hunk 2 above)
Comment (Quadry01) [reply app/LoginPage/page.js:23]: I will work on it sir

File: app/LoginPage/page.js (hunk 3 above)
Comment (Quadry01) [reply app/LoginPage/page.js:19]: i added it when i was working on the Firebase authentication...so the botton will not refresh the page sir

File: app/ProductPage/page.js (hunk 6 above)
Comment (Quadry01) [reply app/ProductPage/page.js:45]: ok sir

File: app/ProductPage/page.js (hunk 7 above)
Comment (Quadry01) [reply app/ProductPage/page.js:71]: ok sir
